With GCC 4.5.0 and GLIBC 2.12, problems occur related to the
machine architecture (with or without setting -march in CFLAGS),
when assembling s_frexp.S, because the preprocessor replaces
__i686 with "1".

See e.g.:

http://www.eglibc.org/archives/patches/msg00073.html

---
 nptl/sysdeps/pthread/pt-initfini.c    |    5 +++++
 sysdeps/unix/sysv/linux/i386/sysdep.h |    4 ++++
 2 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/nptl/sysdeps/pthread/pt-initfini.c b/nptl/sysdeps/pthread/pt-initfini.c
index 9c00dc0..f5d4df8 100644
--- a/nptl/sysdeps/pthread/pt-initfini.c
+++ b/nptl/sysdeps/pthread/pt-initfini.c
@@ -45,6 +45,11 @@
 /* Embed an #include to pull in the alignment and .end directives. */
 asm ("\n#include \"defs.h\"");
 
+asm ("\n#if defined __i686 && defined __ASSEMBLER__");
+asm ("\n#undef __i686");
+asm ("\n#define __i686 __i686");
+asm ("\n#endif");
+
 /* The initial common code ends here. */
 asm ("\n/*@HEADER_ENDS*/");
 
diff --git a/sysdeps/unix/sysv/linux/i386/sysdep.h b/sysdeps/unix/sysv/linux/i386/sysdep.h
index 77e6bad..9ed0b71 100644
--- a/sysdeps/unix/sysv/linux/i386/sysdep.h
+++ b/sysdeps/unix/sysv/linux/i386/sysdep.h
@@ -29,6 +29,10 @@
 #include <dl-sysdep.h>
 #include <tls.h>
 
+#if defined __i686 && defined __ASSEMBLER__
+#undef __i686
+#define __i686 __i686
+#endif
 
 /* For Linux we can use the system call table in the header file
 	/usr/include/asm/unistd.h
-- 
1.6.3.3

